workflows_core/engine/abstract_engine.py

```diff
@@ -100,13 +100,14 @@ def size(self) -> int:
         return self._size
 
     @abstractmethod
-    def apply(self) -> Any:
+    def apply(self) -> float:
         raise NotImplementedError
 
     def __call__(self) -> Any:
         self.operator.pre_hooks(self._dataset)
-        self.apply()
+        success_ratio = self.apply()
```

Comment:
I don't think this should return success ratio. Success ratio should be stored as an internal attribute so that it can be referenced in other areas. 
Returning this is very unintuitive.